api/envoy/api/v2/auth/cert.proto

```diff
@@ -272,12 +272,17 @@ message UpstreamTlsContext {
   // SNI string to use when creating TLS backend connections.
   string sni = 2 [(validate.rules).string.max_bytes = 255];
 
+  // If true, the SNI from the downstream connection is forwarded to the upstream connection.
+  //
+  // TODO: handle oneof with the sni
+  bool forward_original_sni = 3;
```

Comment:
why do you need an option for this? why can't you always forward the downstream SNI if there is no SNI value specified in the cluster?

Reply:
This is what @lizan wanted https://github.com/envoyproxy/envoy/issues/4076#issuecomment-417462072

> Add explicit field forward_original_sni would be better than defaulting to forward.

I proposed:

> Change tcp_proxy to set the SNI of the upstream connection in case the cluster's tls context does not have sni field.

@rshriram @lizan Could you please agree on the API ?

Reply:
Let's be explicit between forward downstream SNI vs no SNI. You should be able to use `oneof` without breaking wire format.